File: ABC/ABC301-400/ABC339/B/B.nim
import strutils, sequtils
let h_w_n = stdin.readLine.split().map(parseInt)
let
  H = h_w_n[0]
  W = h_w_n[1]
  N = h_w_n[2]
var
  A = newSeqWith(H, '.'.repeat(W))
  nowx = 0
  nowy = 0
  dx = -1
  dy = 0
for i in 0..<N:
  if A[nowx][nowy] == '.':
    (dx, dy) = (dy, -dx)
    A[nowx][nowy] = '#'
  else:
    (dx, dy) = (-dy, dx)
    A[nowx][nowy] = '.'
  nowx += dx
  nowy += dy
  nowx = (nowx + H) mod H
  nowy = (nowy + W) mod W
for i in A:
  echo i

File: ABC/ABC301-400/ABC339/B/B.py
H, W, N = map(int, input().split())
A = [['.'] * W for _ in range(H)]
nowx, nowy = 0, 0
dx, dy = -1, 0
for _ in range(N):
	if A[nowx][nowy] == '.':
		dx, dy = dy, -dx
		A[nowx][nowy] = '#'
	else:
		dx, dy = -dy, dx
		A[nowx][nowy] = '.'
	nowx += dx
	nowy += dy
	nowx %= H
	nowy %= W
for i in A:
	print(''.join(i))